src/site/topic6.rst

```diff
@@ -218,8 +218,122 @@ Pop and Peek
 * What I can do however is throw an exception to let the user know something exceptional happened
 * It is up to them to deal with the situation
 
+.. warning::
+
+    To use the ``EmptyStackException``, we will need to import it --- ``import java.util.EmptyStackException;``
+
+size and isEmpty
+^^^^^^^^^^^^^^^^
+
+.. code-block:: Java
+    :linenos:
+    :emphasize-lines: 2
+
+        public int size() {
+            return top;
+        }
+
+        public boolean isEmpty() {
+            return size() == 0;
+        }
+
+* Notice how, because of zero based indexing
+    * ``top`` tells us the next available spot in the array
+    * And the number of things in the stack
+
+toString
+^^^^^^^^
+
+.. code-block:: Java
+    :linenos:
+
+    public String toString() {
+        StringBuilder builder = new StringBuilder();
+        for (int i = 0; i < top; ++i) {
+            builder.append(stack[i]);
+            builder.append(", ");
+        }
+        builder.append("<-- Top\n");
+        return builder.toString();
+    }
+
+* See if you can figure out what the toString will return based on this code
+
+
 For next time
 =============
 
+* Download and test the :download:`Stack <../src/main/java/Stack.java>` and  :download:`ArrayStack <../src/main/java/ArrayStack.java>` code
```

Comment:
Might be dead links (build didn't run since you're not merging into main).

Reply:
Good eye.